hyper-parameter-optimization.ipynb

```diff
@@ -0,0 +1,180 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get resources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "import dask.distributed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cluster = coiled.Cluster(\n",
+    "    n_workers=20, \n",
+    "    configuration=\"coiled/default\", \n",
+    ")\n",
+    "client = dask.distributed.Client(cluster)\n",
+    "\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get and pre-process data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask.dataframe as dd\n",
+    "df = dd.read_csv(\n",
+    "    \"s3://nyc-tlc/trip data/yellow_tripdata_2019-*.csv\", \n",
+    "    parse_dates=[\"tpep_pickup_datetime\", \"tpep_dropoff_datetime\"],\n",
+    "    dtype={\n",
+    "        \"VendorID\": \"UInt8\",\n",
+    "        \"passenger_count\": \"UInt8\",\n",
+    "        \"RatecodeID\": \"UInt8\",\n",
+    "        \"store_and_fwd_flag\": \"category\",\n",
+    "        \"PULocationID\": \"UInt16\",\n",
+    "        \"DOLocationID\": \"UInt16\",    \n",
+    "        \"payment_type\": \"UInt8\",\n",
+    "    },\n",
+    "    blocksize=\"16 MiB\",\n",
+    ")\n",
+    "\n",
+    "data = df[[\"passenger_count\", \"trip_distance\", \"RatecodeID\", \"payment_type\", \"fare_amount\"]]\n",
+    "data = data.fillna(0)\n",
+    "\n",
+    "labels = (df.tip_amount / df.fare_amount) > 0.25\n",
```

Comment:
Nit: I might predict taxi trip duration to mirror https://www.kaggle.com/c/nyc-taxi-trip-duration/. That would imply a regression problem, not a classification problem.

Reply:
Oh cool.  It would be nice to reflect an existing Kaggle problem.

Reply:
And maybe at the end show how many minutes you are off:

``` python
pred_time = model.score(X_test)
err = np.abs(pred_time - real_time)
pd.Series(err).plot.hist()
```